beater/beater.go

```diff
@@ -3,37 +3,37 @@ package beater
 import (
 	"fmt"
 
-	"github.com/elastic/apm-server/config"
 	"github.com/elastic/apm-server/server"
 	"github.com/elastic/beats/libbeat/beat"
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/logp"
 )
 
-type ApmServer struct {
+type ApmBeater struct {
 	done   chan struct{}
-	server *server.Server
-	config config.Config
+	server *server.ApmServer
+	config BeaterConfig
 	client beat.Client
 }
 
 // Creates beater
-func New(_ *beat.Beat, cfg *common.Config) (beat.Beater, error) {
-	config := config.DefaultConfig
-	if err := cfg.Unpack(&config); err != nil {
+func New(_ *beat.Beat, ucfg *common.Config) (beat.Beater, error) {
```

Comment:
`cfg` is kind of an unwritten convention for the `common.Config`. I think we don't use ucfg to not conflict with the package name.